qi-lib/flow/core/compiler.rkt

```diff
@@ -14,14 +14,124 @@
          racket/undefined
          (prefix-in fancy: fancy-app))
 
+(define-syntax inline-compose1
+  (syntax-rules ()
+    [(_ f) f]
+    [(_ f1 f ...) (f1 (inline-compose1 f ...))]))
```

Comment:
Maybe I'm not understanding the usage, but it seems like all the `f`s are functions… so wouldn't `(inline-compose1 foo bar baz)` expand to `(foo (bar (baz)))` which is not the same as `(lambda (x) (foo (bar (baz x)))` that `compose1` generates at runtime?

Reply:
I guess it's possible you've arranged for the arguments to the macro to make more sense in the former arrangement than the latter, in which case I would suggest some comments with examples and choosing a different name.

Reply:
I think the first case would be `(foo (bar baz))`, so, the last parameter is passed as an argument, which, in this case it is itself a function. There's a lot of currying and continuation-passing magic going on here and I confess I can't make it out either, but it does seem as though we might be losing the `f` in each `map` and `filter` since those don't seem to be used in the expansion. I'll take a closer look tonight or we'll aim to annotate with comments live tomorrow!

Reply:
Well, you're right about `(foo (bar baz))` at least :)